web/src/main/java/com/bazaarvoice/emodb/web/scanner/ScanUploader.java

```diff
@@ -58,16 +60,21 @@
     private final StashStateListener _stashStateListener;
     private final CompactionControlSource _compactionControlSource;
     private final DataCenters _dataCenters;
+    private final MetricRegistry _metricRegistry;
+    private final Counter _compactionControlTimeCounter;
 
     @Inject
     public ScanUploader(DataTools dataTools, ScanWorkflow scanWorkflow, ScanStatusDAO scanStatusDAO,
-                        StashStateListener stashStateListener, @DelegateCompactionControl CompactionControlSource compactionControlSource, DataCenters dataCenters) {
+                        StashStateListener stashStateListener, @DelegateCompactionControl CompactionControlSource compactionControlSource, DataCenters dataCenters,
+                        final MetricRegistry metricRegistry) {
         _dataTools = checkNotNull(dataTools, "dataTools");
         _scanWorkflow = checkNotNull(scanWorkflow, "scanWorkflow");
         _scanStatusDAO = checkNotNull(scanStatusDAO, "scanStatusDAO");
         _stashStateListener = checkNotNull(stashStateListener, "stashStateListener");
         _compactionControlSource = checkNotNull(compactionControlSource, "compactionControlSource");
         _dataCenters = checkNotNull(dataCenters, "dataCenters");
+        _metricRegistry = checkNotNull(metricRegistry, "metricRegistry");
```

Comment:
Unnecessary creation of class attribute, `metricRegistry` is only used in this method.